src/tqec/gallery/cz.py

```diff
@@ -91,33 +92,17 @@ def _resolve_ports(flows: list[str]) -> list[ZXKind]:
     for i in range(4):
         if ports[i] == "_":
             ports[i] = "Z"
-    # note that node kind is opposite to the supported observable basis
-    return [ZXKind(p).with_zx_flipped() for p in ports]
-
 
-def logical_cz_block_graph(support_flows: str | list[str] | None = None) -> BlockGraph:
-    """Create a block graph for the logical CZ gate.
+    def _vkind(p: str) -> ZXCube:
+        if p == "Z":
+            return ZXCube.from_str("XZZ")
+        else:
+            return ZXCube.from_str("XZX")
 
-    By default, the Hadamard edge in the CZ gate will be horizontal. If you
-    want to use vertical Hadamard edges, you can rotate the graph by 90 degrees
-    by calling `~tqec.computation.block_graph.BlockGraph.rotate`.
+    def _hkind(p: str) -> ZXCube:
+        if p == "Z":
+            return ZXCube.from_str("XZZ")
+        else:
+            return ZXCube.from_str("XXZ")
```

Comment:
If I understood correctly, `p` can only be `X` or `Z`, and so these can likely be replaced by
```py
def _vkind(p: str) -> ZXCube:
    return ZXCube.from_str(f"XZ{p}")
def _hkind(p: str) -> ZXCube:
    return ZXCube.from_str(f"X{p}Z")
```